streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -63,10 +66,21 @@ public RecordCollector(Producer<byte[], byte[]> producer) {
         this.offsets = new HashMap<>();
     }
 
-    public <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
+    public void send(ProducerRecord<byte[], byte[]> record) {
+        this.producer.send(record, callback);
+    }
+
+    public <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer,
```

Comment:
nit: we can keep the send() call without the partitioner argument which will then call this function with null.